addons/mail/models/mail_channel.py

```diff
@@ -746,6 +747,36 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def toggle_message_pin(self, message_id, pinned):
```

Comment:
As you give the value of is_pinned, it is not really a toggle but a set. Toggle is generally used without arguments (to toggle a state).